src/app/components/turnos/gestor-agendas/operaciones-turnos/reasignar/reasignar-turno-agendas.component.ts

```diff
@@ -75,10 +88,35 @@ export class ReasignarTurnoAgendasComponent implements OnInit {
     }
 
     seleccionarCandidata(indiceTurno, indiceBloque, indiceAgenda) {
-        const turno = this.agendasSimilares[indiceAgenda].bloques[indiceBloque].turnos[indiceTurno];
-        const turnoSiguiente = this.agendasSimilares[indiceAgenda].bloques[indiceBloque].turnos[indiceTurno + 1];
-        const bloque = this.agendasSimilares[indiceAgenda].bloques[indiceBloque];
-        this.agendaSeleccionada = this.agendasSimilares[indiceAgenda];
+        if (this.indiceTurno === indiceTurno && this.indiceBloque === indiceBloque && this.indiceAgenda === indiceAgenda) {
+            this.indiceTurno = null;
+            this.indiceBloque = null;
+            this.indiceAgenda = null;
+            this.turnoSel = null;
+            this.turnoSiguiente = null;
+            this.bloqueSel = null;
+            this.agendaSeleccionada = null;
+            this.seleccionoHorario = false;
+        } else {
+            this.indiceTurno = indiceTurno;
+            this.indiceBloque = indiceBloque;
+            this.indiceAgenda = indiceAgenda;
+            this.turnoSel = this.agendasSimilares[indiceAgenda].bloques[indiceBloque].turnos[indiceTurno];
+            this.turnoSiguiente = this.agendasSimilares[indiceAgenda].bloques[indiceBloque].turnos[indiceTurno + 1];
+            this.bloqueSel = this.agendasSimilares[indiceAgenda].bloques[indiceBloque];
+            this.agendaSeleccionada = this.agendasSimilares[indiceAgenda];
+            this.seleccionoHorario = true;
+        }
+    }
+
+    horarioSeleccionado(indiceTurno, indiceBloque, indiceAgenda) {
+        if (indiceTurno === this.indiceTurno && indiceBloque === this.indiceBloque && indiceAgenda === this.indiceAgenda) {
+            return true;
+        }
+        return false;
```

Comment:
Esto se puede simplificar con una ternaria:
return (indiceTurno === this.indiceTurno && ...) ? true : false